### PR TITLE
refactor: removing unused variable

### PR DESCRIPTION
**Pull request by imprateeksh**
### Description
**Enhancement**: To remove the unused optional variable `network_connections`
Refer [issue 485](https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone-vpc/issues/485) for details.

#### No release required

- [ ] Examples or tests (addition or updates of examples or tests)
- [x] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [] Other changes that don't affect Terraform code

#### Release required

- [ ] Bug fix (patch release (`x.x.X`): Change that fixes an issue and is compatible with earlier versions)
- [ ] New feature (minor release (`x.X.x`): Change that adds functionality and is compatible with earlier versions)
- [x] Breaking change (major release (`X.x.x`): Change that is likely incompatible with previous versions)

##### Release notes content
fix: The `network_connections` option has been removed from the `network_acls` variable as it was not being used in the code

BREAKING CHANGE: If you are upgrading to this version, and you have the `network_connections` option in the `network_acls` variable, it should be removed.

---

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [x] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Merge by using "Squash and merge".
- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author.

    The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).

### CODE REVIEW

**Comment by imprateeksh** [comment]
This passed successfully but once integrated latest main changes today, can see some failure in the upgrade test.
It appears that network ACL rule already exists.
image

Re-running first to see if this error still occurs.

**Comment by imprateeksh** [comment]
Validation succeeded after re-running the jobs.

File: variables.tf [hunk 1]
```diff
@@ -96,11 +96,10 @@ variable "address_prefixes" {
 ##############################################################################
 
 variable "network_acls" {
-  description = "List of ACLs to create. Rules can be automatically created to allow inbound and outbound traffic from a VPC tier by adding the name of that tier to the `network_connections` list. Rules automatically generated by these network connections will be added at the beginning of a list, and will be web-tierlied to traffic first. At least one rule must be provided for each ACL."
+  description = "List of ACLs to create.At least one rule must be provided for each ACL."
```
Comment (ocofaigh) [inline variables.tf:99]: missing a space after the full stop

**Comment by ocofaigh** [comment]
@imprateeksh why have you ticked the boxes for `Documentation update` and `Other changes that don't affect Terraform code` ? This is a change to a terraform variable. Infact its probably a breaking change which will warrant a major version update. Don't you agree?

File: module-metadata.json [hunk 2]
```diff
@@ -157,8 +157,8 @@
...
+      "description": "List of ACLs to create.At least one rule must be provided for each ACL.",
```
Comment (SirSpidey) [inline module-metadata.json:161]:
```suggestion
      "description": "The list of ACLs to create. Provide at least one rule for each ACL.",
```

**Review by SirSpidey — COMMENTED** [review]
small suggestion

File: module-metadata.json (hunk 2 above)
Comment (imprateeksh) [reply module-metadata.json:161]: Thank you @SirSpidey for your observation. I have made this change.

File: variables.tf (hunk 1 above)
Comment (imprateeksh) [reply variables.tf:99]: Sure. This is now addressed.

**Comment by imprateeksh** [comment]
> @imprateeksh why have you ticked the boxes for `Documentation update` and `Other changes that don't affect Terraform code` ? This is a change to a terraform variable. Infact its probably a breaking change which will warrant a major version update. Don't you agree?

HI @ocofaigh - `Documentation update` was chosen as there was change in the description of the variable. Indeed it is breaking change and `Other changes that don't affect Terraform code` is not applicable here. 
I have made these corrections.

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**